pkg/client/googlegroup_github/mapper.go

```diff
@@ -19,15 +19,88 @@ import (
 	"context"
 	"fmt"
 	"os"
+	"slices"
 	"strings"
 
 	"google.golang.org/protobuf/encoding/prototext"
 
 	"github.com/abcxyz/pkg/logging"
 	"github.com/abcxyz/team-link/apis/v1alpha3"
+	"github.com/abcxyz/team-link/pkg/github"
 	"github.com/abcxyz/team-link/pkg/groupsync"
 )
 
+// GroupMapper implements groupsync.OneToManyGroupMapper
+// For now it was be used as Mapper between GoogleGroup
+// and GitHubTeam.
+// Both GoogleGroupMapper and GitHubMapper have similar
+// variables and functions, this way we don't need to repeatly
+// write similar function for both on them.
+type GroupMapper struct {
+	mappings map[string][]string
+}
+
+func (m *GroupMapper) AllGroupIDs(ctx context.Context) ([]string, error) {
+	res := make([]string, 0)
+	for key := range m.mappings {
+		res = append(res, key)
+	}
+	slices.Sort(res)
+	return res, nil
+}
+
+func (m GroupMapper) ContainsGroupID(ctx context.Context, key string) (bool, error) {
+	_, ok := m.mappings[key]
+	if !ok {
+		return false, fmt.Errorf("group %s is not mapped", key)
```

Comment:
Why would this return an error? It seems like we always want to return a boolean?

Reply:
Same, it's how the interface is defined.

Reply:
But this is just checking whether something exists in a map. This shouldn't return an _error_ when the entry is not found: it should return `false, nil`. Otherwise you're overloading the return values.

Reply:
Oh, I think I misunderstood your point, I will update this now

Reply:
Updated, thank you!